pulsar-client-cpp/lib/ProducerImpl.cc

```diff
@@ -856,7 +856,18 @@ bool ProducerImpl::ackReceived(uint64_t sequenceId, MessageId& rawMessageId) {
         lastSequenceIdPublished_ = sequenceId + op.messagesCount_ - 1;
 
         pendingMessagesQueue_.pop_front();
-
+        int totalChunks = op.metadata_.num_chunks_from_msg();
+        if (totalChunks > 1) {
+            if (op.metadata_.chunk_id() == 0) {
+                op.chunkedMessageId_ = MessageId(partition_, rawMessageId.ledgerId(), rawMessageId.entryId(),
+                                                 rawMessageId.batchIndex());
+            } else if (op.metadata_.chunk_id() == totalChunks - 1) {
+                MessageId lastMessageId(partition_, rawMessageId.ledgerId(), rawMessageId.entryId(),
+                                        rawMessageId.batchIndex());
+                op.chunkedMessageId_ = MessageId(op.chunkedMessageId_, lastMessageId);
```

Comment:
We have already created a message ID with the correct partition index, see the first few lines in `ackReceived`.

```c++
    MessageId messageId(partition_, rawMessageId.ledgerId(), rawMessageId.entryId(),
                        rawMessageId.batchIndex());
```

Could we simplify the code to

```c++
            if (op.metadata_.chunk_id() == 0) {
                op.chunkedMessageId_ = messageId;
            } else if (op.metadata_.chunk_id() == totalChunks - 1) {
                op.chunkedMessageId_ = MessageId(op.chunkedMessageId_, messageId);
                messageId = op.chunkedMessageId_;
            }
```